django-s3-file-field-client/src/client.ts

```diff
@@ -80,7 +94,7 @@ export default class S3FFClient {
     // upload each part of the buffer in parallel using the calculated indices
     return await Promise.all(parts.map(async (part, i) => {
       const chunk = buffer.slice(indices[i], indices[i] + part.size);
-      return await this.uploadPart(chunk, part);
+      return await this.uploadPart(chunk, part, options);
```

Comment:
These functions are called in parallel. Each one will be competing to concurrently fire `ProgressEvent` (with different contexts) back at the single callback.

Reply:
I was prepared to keep this library very dumb and do the accumulation myself, but then we may end up wanting to change the behavior to be more helpful in the future. I can just go ahead and change it now so that this library only issues cumulative progress events rather than issuing them for each chunk independently.